api/src/Entity/ContentNode/MultiSelect.php

```diff
@@ -37,59 +36,39 @@
     normalizationContext: ['groups' => ['read']],
 )]
 #[ORM\Entity(repositoryClass: MultiSelectRepository::class)]
-#[ORM\Table(name: 'content_node_multiselect')]
 class MultiSelect extends ContentNode {
-    #[ApiProperty(readableLink: true, writableLink: false)]
-    #[Groups(['read'])]
-    #[ORM\OneToMany(targetEntity: MultiSelectOption::class, mappedBy: 'multiSelect', orphanRemoval: true, cascade: ['persist'])]
-    public Collection $options;
-
-    public function __construct() {
-        parent::__construct();
-        $this->options = new ArrayCollection();
-
-        parent::__construct();
-    }
-
-    /**
-     * @return MultiSelectOption[]
-     */
-    public function getOptions(): array {
-        return $this->options->getValues();
-    }
-
-    public function addOption(MultiSelectOption $option): self {
-        if (!$this->options->contains($option)) {
-            $this->options->add($option);
-            $option->multiSelect = $this;
-        }
-
-        return $this;
-    }
-
-    public function removeOption(MultiSelectOption $option): self {
-        if ($this->options->removeElement($option)) {
-            if ($option->multiSelect === $this) {
-                $option->multiSelect = null;
-            }
-        }
-
-        return $this;
-    }
+    public const JSON_SCHEMA = [
+        'type' => 'object',
+        'additionalProperties' => false,
+        'required' => ['options'],
+        'properties' => [
+            'options' => [
+                'type' => 'object',
+                'additionalProperties' => ['$ref' => '#/$defs/option'],
+            ],
+        ],
+        '$defs' => [
+            'option' => [
+                'type' => 'object',
+                'additionalProperties' => false,
+                'required' => ['checked'],
+                'properties' => [
+                    'checked' => [
+                        'type' => 'boolean',
+                    ],
+                ],
+            ],
+        ],
+    ];
 
     /**
-     * @param MultiSelect $prototype
-     * @param EntityMap   $entityMap
+     * Holds the actual data of the content node
+     * (overridden from abstract class in order to add specific validation).
      */
-    public function copyFromPrototype($prototype, $entityMap): void {
-        parent::copyFromPrototype($prototype, $entityMap);
-
-        // copy all multiSelect options
-        foreach ($prototype->options as $optionPrototype) {
-            $option = new MultiSelectOption();
-            $this->addOption($option);
-
-            $option->copyFromPrototype($optionPrototype, $entityMap);
-        }
-    }
+    #[ApiProperty(example: null)]
```

Comment:
Könnten wir hier trotzdem ein example einfügen? Mit dem JSON data Feld ists viel intransparenter in der Dokumentation, was überhaupt erlaubt sein könnte, und so ein Example könnte bei Get oder besonders Patch helfen.

Reply:
Yes, macht Sinn. Muss dann aber noch die Tests anpassen, weil dort wird der default-Payload anhand der example values zusammengesetzt.

Reply:
Fixed in https://github.com/ecamp/ecamp3/pull/2825/commits/dd67ee301305c9596c196ba0562d973d13fb0c69